migrations/V16__create_reports.sql

```diff
@@ -0,0 +1,35 @@
+CREATE TYPE t_pos_neg_nil AS ENUM ('POSITIVE', 'NEGATIVE', 'NIL');
+CREATE TYPE t_reported_ineligibility AS ENUM ('NIL', 'TEMPORARY_INELIGIBLE', 'PERMANENTLY_INELIGIBLE');
+
+CREATE TABLE reports (
+  report_id BIGSERIAL,
+  report_creation_time TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  report_modification_time TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  vet_id BIGINT NOT NULL,
+  dog_id BIGINT NOT NULL,
```

Comment:
We don't need these relations. Transitively we can get to reports via call from both dogs and vets.

```
dogs <- calls(call_outcome='APPOINTMENT') <- reports
vets <- calls(call_outcome='APPOINTMENT') <- reports
```

Reply:
Reasons why we will keep the `dog_id` and `vet_id` FKs.

—(a) Having the FKs makes it more efficient to retrieve reports related to dogs and vets. We are going to do this a lot. (Some experiment results are shown in Fig1 and Fig2.)

—(b) Conceptually, we think of reports being associated with dogs and vets more than we think of them being associated with calls. The call relation is a little extra that's nice to have.

—(c) The above "concept" is validated by future designs in Figma whereby vets can create medical reports without an appointment. When we do support that, `call_id` might even be nullable.

___

Fig1: Query & Plan when using calls table as an intermediary to select reports related to a specific dog.
```
create index if not exists calls_dog_idx on calls (dog_id);
create index if not exists reports_call_idx on reports (call_id);
EXPLAIN
SELECT * FROM reports WHERE call_id IN (
    SELECT call_id
    FROM calls
    WHERE dog_id = '6'
);

Hash Join  (cost=12.69..29.77 rows=3 width=116)
  Hash Cond: (reports.call_id = calls.call_id)
  ->  Seq Scan on reports  (cost=0.00..15.60 rows=560 width=116)
  ->  Hash  (cost=12.64..12.64 rows=4 width=8)
        ->  Bitmap Heap Scan on calls  (cost=4.18..12.64 rows=4 width=8)
              Recheck Cond: (dog_id = '6'::bigint)
              ->  Bitmap Index Scan on calls_dog_idx  (cost=0.00..4.18 rows=4 width=0)
                    Index Cond: (dog_id = '6'::bigint)
```

___

Fig2: Query & Plan when selecting reports related to a specific dog using the `dog_id` FK directly.

```
create index if not exists reports_dog_idx on reports (dog_id);
EXPLAIN
SELECT * FROM reports WHERE dog_id = '6';

Bitmap Heap Scan on reports  (cost=4.17..11.28 rows=3 width=116)
  Recheck Cond: (dog_id = '6'::bigint)
  ->  Bitmap Index Scan on reports_dog_idx  (cost=0.00..4.17 rows=3 width=0)
        Index Cond: (dog_id = '6'::bigint)
```

Reply:
Reasons why we should drop redundant FKs

—(a) Avoids the possibility of incoherent references whereby the referenced vet or dog is different from what the referenced call believes.

—(b) Minor point. Storage efficiency.

Usability concerns can be managed by adding a view to pull in the dog and vet IDs.

Reply:
Okay. I'm convinced. Let's drop the FKs.

Reply:
FKs removed - 126fa0f1827ecba69e16efc527fa39dfe4f54faa